pkg/k8sclient/nodewatcher.go

```diff
@@ -268,10 +268,11 @@ func (nw *NodeWatcher) nodeWorker() {
 				case NodeUpdated:
 					NodeMux.RLock()
 					rtnd, ok := NodeToRTND[node.Hostname]
-					NodeMux.RUnlock()
 					if !ok {
 						glog.Fatalf("Node %s does not exist", node.Hostname)
```

Comment:
If node does not exist and then the process exit, do we still have chance to release the read lock?

Reply:
Move `NodeMux.RUnlock()` up and place it around 

```
rtnd, ok := NodeToRTND[node.Hostname]
```

will fix this issue?

Reply:
@m1093782566 if a process fatal's out we don't need to worry about releasing locks or graceful shutdown.

Reply:
Yes, I just want a clarification :) But, I still want something looks like:

```
NodeMux.RLock()
rtnd, ok := NodeToRTND[node.Hostname]
NodeMux.RULock()
```

In case we change glog.Fatal to glog.Errorf some day.

Reply:
@m1093782566 I think we can fix those cases when we do the HA enhancement for Poseidon.
Let it crash/fatal out for time being this way we will come to know where it failing.